tests/sflow/test_sflow.py

```diff
@@ -395,63 +455,70 @@ def testDifferntPollingInt(self, duthost, partial_ptf_runner):
 class TestSflowInterface():
     """
     Enable / Disable Sflow interfaces and check the samples are received only from  the intended interface
-    Test interfaceswith different sampling rates
+    Test interfaces with different sampling rates
     """
 
-    def testIntfRemoval(self, sflowbase_config, duthost, partial_ptf_runner):
-        sflow_int = sorted(var['sflow_ports'].keys())
-        config_sflow_interfaces(duthost, sflow_int[0], status='disable')
-        config_sflow_interfaces(duthost, sflow_int[1], status='disable')
+    def testIntfRemoval(self, sflowbase_config, duthost, partial_ptf_runner, tested_portchannel_members,
+                        restore_sflow_interface_status_and_rate):
+        disabled_sflow_intf_list = tested_portchannel_members[0]
+        enabled_sflow_intf_list = [intf for intf_list in tested_portchannel_members[1:] for intf in intf_list]
+
+        # Disable sflow for the interface in first portchannel
+        for intf in disabled_sflow_intf_list:
+            config_sflow_interfaces(duthost, intf, status='disable')
+            verify_sflow_interfaces(duthost, intf, 'down', SFLOW_RATE_DEFAULT)
 
-        verify_sflow_interfaces(duthost, sflow_int[0], 'down', 512)
-        verify_sflow_interfaces(duthost, sflow_int[1], 'down', 512)
-        enabled_intf = sflow_int[2:]
-        for intf in enabled_intf:
-            verify_sflow_interfaces(duthost, intf, 'up', 512)
+        # Enable sflow for the interface in other portchannel
+        for intf in enabled_sflow_intf_list:
+            verify_sflow_interfaces(duthost, intf, 'up', SFLOW_RATE_DEFAULT)
+
+        # Traffic test for the enabled sflow interfaces
         partial_ptf_runner(
-            enabled_sflow_interfaces=enabled_intf,
+            enabled_sflow_interfaces=enabled_sflow_intf_list,
             active_collectors="['collector0','collector1']")
 
-    def testIntfSamplingRate(self, sflowbase_config, duthost, ptfhost, partial_ptf_runner):
+    def testIntfSamplingRate(self, sflowbase_config, duthost, ptfhost, partial_ptf_runner, tested_portchannel_members,
+                             restore_sflow_interface_status_and_rate):
+        first_portchannel_members = tested_portchannel_members[0]
+        second_portchannel_members = tested_portchannel_members[1]
```

Comment:
Can we guarantee that a second physical interface always exists?

Reply:
Add a check for it.
```python
    if len(portchannel_interfaces_list) < 2:
        pytest.skip("The test requires at least two portchannels with multiple members")
```